src/facade/dragonfly_connection.cc

```diff
@@ -1995,6 +1998,10 @@ void Connection::MarkReadBufferConsumed() {
   }
 }
 
+uint32_t& Connection::NumConns() {
+  return IsMain() ? stats_->num_conns_main : stats_->num_conns_other;
+}
```

Comment:
I really do not like to return references especially when we only want to increment what is being referenced. I know that Stefan suggested this but I would also suggest a slightly change: Rename `NumConns()` to `IncrNumConns()` and do the increment here:

```
void Connection::IncrNumConnStats() {
  IsMain() ? ++stats_->blah blah....
}
```

Reply:
changed